app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -260,6 +260,11 @@ const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(supercla
             batch.push(...postFiles);
         }
 
+        const allFiles = await database.get<FileModel>(MM_TABLES.SERVER.FILE).query(Q.where('post_id', Q.oneOf(uniquePosts.map((p) => p.id)))).fetch();
+        const receivedFilesSet = new Set(files.map((f) => f.id));
+        const removedFiles = allFiles.filter((f) => !receivedFilesSet.has(f.id)).map((f) => f.prepareDestroyPermanently());
```

Comment:
Would it be better to use reduce here? Not to mentioned, you can add batch.push within the reduce() as well so you don't have to spread the array?

Reply:
Changed it to a forEach instead